### PR TITLE
Update deps and included dependabot settings for action

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Updated the deps for action
- Included the dependabot settings for detect-workflow

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]

**Comment by ianlewis** [comment]
There seems to be an issue with the test OIDC server that only started showing up when updating here. It's causing tests to fail. I'll fix in another PR later if someone doesn't get to it before me.

**Review by ianlewis — APPROVED** [review]

**Comment by laurentsimon** [comment]
now that we've enabled renovatebot for a trial, shall we close this issue?

**Comment by naveensrinivasan** [comment]
> If @naveensrinivasan can merge and the changes to dependabot.yml are taken out I'm ok with the other changes.

I can remove the dependabot.yml

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,6 @@ vendor/
 # Go workspace file
 go.work
 go.work.sum
+
+# goland
+.idea
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply .gitignore]: Took care of it.

File: .github/actions/detect-workflow/main_test.go [hunk 2]
```diff
@@ -106,9 +106,12 @@ func Test_action_getRepoRef(t *testing.T) {
...
 		s, c := github.NewTestOIDCServer(t, now, &github.OIDCToken{
-			Audience:       []string{"githubuser/reponame/detect-workflow"},
-			Expiry:         time.Date(2022, 5, 4, 0, 0, 0, 0, time.UTC),
-			JobWorkflowRef: "githubuser/reponame/path/to/workflow@refs/heads/main",
+			Audience:          []string{"githubuser/reponame/detect-workflow"},
+			Expiry:            time.Date(2022, 5, 4, 0, 0, 0, 0, time.UTC),
+			JobWorkflowRef:    "githubuser/reponame/path/to/workflow@refs/heads/main",
+			RepositoryOwnerID: "githubuser",
+			ActorID:           "githubuser",
+			RepositoryID:      "reponame",
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply .github/actions/detect-workflow/main_test.go]: Updated it. Thanks

**Pull request merged**